### PR TITLE
(fix) O3-3970 Re-enable readonly fields

### PR DESCRIPTION
**Pull request by pirupius**
## Requirements

- [x] This PR has a title that briefly describes the work done including the ticket number. If there is a ticket, make sure your PR title includes a [conventional commit](https://o3-dev.docs.openmrs.org/#/getting_started/contributing?id=your-pr-title-should-indicate-the-type-of-change-it-is) label. See existing PR titles for inspiration.
- [x] My work conforms to the [OpenMRS 3.0 Styleguide](https://om.rs/styleguide) and [design documentation](https://zeroheight.com/23a080e38/p/880723-introduction).
- [x] My work includes tests or is validated by existing tests.

## Summary
This PR enables readonly mode in the field types that might be having it as well as adding the utility to cater for `boolean` string values

**NOTE**: For the date, we're using `OpenmrsDatePicker` and the readonly still shows the date picker so this might be either in newer versions of the framework or needs to be implemented in `esm-core`

## Screenshots

Text
Screenshot 2024-09-26 at 13 35 25

Multi Select
Screenshot 2024-09-27 at 13 33 27

Number
Screenshot 2024-09-26 at 13 35 37

Radio
Screenshot 2024-09-26 at 13 35 51

Dropdown
Screenshot 2024-09-26 at 13 57 51

Textarea
Screenshot 2024-09-26 at 13 59 13

Toogle
Screenshot 2024-09-26 at 14 03 14

Ui-select extended
Screenshot 2024-09-26 at 14 05 03

Workspace
Screenshot 2024-09-26 at 14 08 00

## Related Issue
https://openmrs.atlassian.net/browse/O3-3970

### CODE REVIEW

**Comment by samuelmale** [comment]
@pirupius Can you attach screenshots or a demo video or tests cases showing the readonly feat in action for the different renderings.

**Comment by pirupius** [comment]
@samuelmale this is ready

File: src/components/inputs/date/date.component.tsx [hunk 1]
```diff
@@ -72,7 +72,7 @@ const DateField: React.FC<FormFieldInputProps> = ({ field, value: dateValue, err
     [field.datePickerFormat, field.label, t],
   );
 
-  return sessionMode == 'view' || sessionMode == 'embedded-view' ? (
+  return sessionMode == 'view' || sessionMode == 'embedded-view' || isTrue(field.readonly) ? (
```
Comment (samuelmale) [inline src/components/inputs/date/date.component.tsx:75]: Shouldn't [this](https://github.com/openmrs/openmrs-esm-form-engine-lib/blob/fea4fb83eb1bad938bcad8f15c04a66543df61ad/src/components/inputs/date/date.component.tsx#L118) and [this](https://github.com/openmrs/openmrs-esm-form-engine-lib/blob/fea4fb83eb1bad938bcad8f15c04a66543df61ad/src/components/inputs/date/date.component.tsx#L121) suffice?

File: src/components/inputs/multi-select/multi-select.component.tsx [hunk 2]
```diff
@@ -80,7 +80,7 @@ const MultiSelect: React.FC<FormFieldInputProps> = ({ field, value, errors, warn
     return field.isRequired ? <FieldLabel field={field} /> : <span>{t(field.label)}</span>;
   }, [field.isRequired, field.label, t]);
 
-  return sessionMode == 'view' || sessionMode == 'embedded-view' ? (
+  return sessionMode == 'view' || sessionMode == 'embedded-view' || isTrue(field.readonly) ? (
```
Comment (samuelmale) [inline src/components/inputs/multi-select/multi-select.component.tsx:83]: Ditto

File: src/components/inputs/date/date.component.tsx (hunk 1 above)
Comment (pirupius) [reply src/components/inputs/date/date.component.tsx:75]: We're currently using the OpenMRS date picker for the date field (calendar) and time picker if specified. Providing the readonly works as intended for the time picker but this doesn't propagate to the OpenMRS date picker as seen in this screenshot. So the fall back for that was to use the view only rendering hence the need for this. 

Screenshot 2024-09-26 at 23 12 29

File: src/components/inputs/multi-select/multi-select.component.tsx (hunk 2 above)
Comment (pirupius) [reply src/components/inputs/multi-select/multi-select.component.tsx:83]: For this component we're using `CheckboxGroup` and it has a finicky behavior where regardless of if that property is passed, the user can still interact with the input field. 

We might have to consider refactoring this component to use `MultiSelect` from Carbon later on that has support for this natively but for now this defaults it to the same display as the view only mode.

**Pull request merged**